components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/OptimizedApplicationConfig.java

```diff
@@ -315,15 +316,21 @@ public AuthenticationStep[] getAuthenticationSteps(String tenantDomain) throws F
         return authenticationSteps;
     }
 
-    private LocalAuthenticatorConfig[] getLocalAuthenticatorConfigs(List<String> localAuthConfigNames) {
+    private LocalAuthenticatorConfig[] getLocalAuthenticatorConfigs(List<String> localAuthConfigNames,
+                                                                    String tenantDomain) throws FrameworkException {
 
-        LocalAuthenticatorConfig[] localAuthenticatorConfigs = new
-                LocalAuthenticatorConfig[localAuthConfigNames.size()];
-        for (int i = 0; i < localAuthConfigNames.size(); i++) {
-            localAuthenticatorConfigs[i] = ApplicationAuthenticatorService.getInstance().
-                    getLocalAuthenticatorByName(localAuthConfigNames.get(i));
+        try {
+            LocalAuthenticatorConfig[] localAuthenticatorConfigs = new
+                    LocalAuthenticatorConfig[localAuthConfigNames.size()];
+            for (int i = 0; i < localAuthConfigNames.size(); i++) {
+                localAuthenticatorConfigs[i] = ApplicationAuthenticatorService.getInstance()
+                        .getLocalAuthenticatorByName(localAuthConfigNames.get(i), tenantDomain);
+            }
+            return localAuthenticatorConfigs;
+        } catch (AuthenticatorMgtException e) {
+            throw new SessionDataStorageOptimizationException(String.format("Error while getting local authenticator " +
```

Comment:
Why we need to use this exception type. I see different exception types are being thrown overall

Reply:
Will resolve this